From e120fe5b6bc03ed5f00de5f3e219329037ec46de Mon Sep 17 00:00:00 2001
From: Leon <ultrafunkamsterdam@users.noreply.github.com>
Date: Sat, 30 Aug 2025 18:03:02 +0200
Subject: [PATCH] Update server.vala

---
 
 diff --git a/server/server.vala b/server/server.vala
 index 593b0811..9ee04707 100644
 --- a/server/server.vala
 +++ b/server/server.vala
 @@ -1,7 +1,7 @@
 namespace Frida.Server {
        private static Application application;

-       private const string DEFAULT_DIRECTORY = null;
+       private static string DEFAULT_DIRECTORY = null ;
        private static bool output_version = false;
        private static string? device_id = null;
        private static string? listen_address = null;
@@ -52,7 +52,8 @@ namespace Frida.Server {
        };

        private static int main (string[] args) {
-               DEFAULT_DIRECTORY = GLib.Uuid.string_random();
+                DEFAULT_DIRECTORY = GLib.Uuid.string_random();
+
                Environment.init ();

 #if DARWIN


diff --git a/src/api/generate.py b/src/api/generate.py
index fd26a800..52643f6c 100644
--- a/src/api/generate.py
+++ b/src/api/generate.py
@@ -247,8 +247,10 @@ def emit_gir(api: ApiSpec, core_gir: str, base_gir: str, output_dir: Path) -> st
     merge_and_transform_elements("class", object_type_names)
     merge_and_transform_elements("interface", object_type_names)
     merge_and_transform_elements("enumeration", enum_type_names | error_type_names)
-
-    ET.indent(merged_root, space="  ")
+    try:
+        ET.indent(merged_root, space="  ")
+    except:
+        pass
     result = ET.tostring(merged_root,
                          encoding="unicode",
                          xml_declaration=True)
